cuiman/src/cuiman/cli/config.py

```diff
@@ -27,38 +31,147 @@ def get_config(config_path: Path | str | None) -> ClientConfig:
+    ctx.curr_params["token"] = login(auth_config)
+    _configure_token_type_with_prompt(ctx)
+
+
+def _configure_token_auth_with_prompt(ctx: _Context) -> None:
+    _prompt_for_str(ctx, "token", "API access token", "")
+    _configure_token_type_with_prompt(ctx)
+
+
+def _configure_api_key_auth_with_prompt(ctx: _Context) -> None:
+    _prompt_for_str(ctx, "api_key", "API access key", "")
+    _prompt_for_str(ctx, "api_key_header", "Access key header", "X-API-Key")
+
+
+def _configure_username_password_with_prompt(ctx: _Context) -> None:
+    _prompt_for_str(
+        ctx,
+        "username",
+        "Username",
+        os.environ.get("USER") or os.environ.get("USERNAME") or "",
+    )
+    _prompt_for_pw(ctx, "password", "Password")
+
+
+def _configure_token_type_with_prompt(ctx: _Context) -> None:
+    use_bearer = _prompt_for_bool(ctx, "use_bearer", "Use bearer token?", False)
+    if not use_bearer:
+        _prompt_for_str(ctx, "token_header", "Access token header", "X-Auth-Token")
+
+
+def _prompt_for_str(
+    ctx: _Context, key: str, text: str, default: str, choice: click.Choice | None = None
+) -> str:
+    value: str | None = ctx.cli_params.get(key)
+    if value is None:
+        value = (
+            typer.prompt(
+                text,
+                type=choice or str,
+                default=ctx.prev_params.get(key) or default,
+            )
+            or ""
         )
-    if not access_token:
-        prev_access_token = config and config.access_token
-        _access_token = typer.prompt(
-            "Access token",
+    ctx.curr_params.update({key: value})
+    assert isinstance(value, str)
+    return value
+
+
+def _prompt_for_pw(
+    ctx: _Context,
+    key: str,
+    text: str,
+) -> str:
+    pw = ctx.cli_params.get(key)
+    if pw is None:
+        prev_pw: str | None = ctx.prev_params.get(key)
+        new_pw: str = typer.prompt(
+            text,
             type=str,
             hide_input=True,
-            default="*****" if prev_access_token else None,
+            default=_HIDDEN_INPUT if prev_pw else None,
         )
-        if _access_token == "*****" and prev_access_token:
-            access_token = prev_access_token
+        if new_pw == _HIDDEN_INPUT and prev_pw:
+            pw = prev_pw
         else:
-            access_token = _access_token
-    if not server_url:
-        server_url = typer.prompt(
-            "Server URL",
-            default=(config and config.server_url) or DEFAULT_SERVER_URL,
+            pw = new_pw
+    ctx.curr_params.update({key: pw})
+    assert isinstance(pw, str)
+    return pw
+
+
+def _prompt_for_bool(
+    ctx: _Context,
+    key: str,
+    text: str,
+    default: bool,
+) -> bool:
+    value: bool | None = ctx.cli_params.get(key)
+    if value is None:
+        value = (
+            typer.confirm(
+                text,
+                default=ctx.prev_params.get(key) or default,
+            )
+            is True
```

Comment:
```suggestion
```

AFAICT `is True` is redundant because `typer.confirm` only ever returns booleans.

Reply:
Fixed!